indico/modules/events/surveys/controllers/display.py

```diff
@@ -84,6 +89,7 @@ def _checkParams(self, params):
                        .options(joinedload('submissions'),
                                 joinedload('sections').joinedload('children'))
                        .one())
+        self.submission = SurveySubmission.find_first(user=session.user, survey=self.survey)
```

Comment:
Actually, even better:

    session.user.survey_submissions.filter_by(survey=survey, is_submitted=False).first()

You don't want to get a "finalized" submission there after all.